include/envoy/http/codec.h

```diff
@@ -142,6 +142,12 @@ class ResponseEncoder : public virtual StreamEncoder {
    * @param trailers supplies the trailers to encode.
    */
   virtual void encodeTrailers(const ResponseTrailerMap& trailers) PURE;
+
+  /**
+   * Indicates whether invalid HTTP messaging should be handled with a stream error or a connection
+   * error.
```

Comment:
What does nullopt mean in this context?

Reply:
It would mean that the new HTTP/1.1 config `stream_error_on_invalid_http_message` hasn't been set, indicating that we need to fall back to the HCM behaviour.

Reply:
OK makes sense can you clarify the comment? Also what does this mean in the case of upstream since there is no HCM?